### PR TITLE
KDR chart

### PR DESCRIPTION
**Pull request by nitzanzuler**
New Kubernetes audit logs webhook + new deployment and service.

### CODE REVIEW

File: wiz-admission-controller/templates/deploymentauditlogs.yaml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- if .Values.podCustomEnvironmentVariablesFile }}
+          - name: CLI_ENV_FILE
+            value: {{ .Values.podCustomEnvironmentVariablesFile }}
+          - name: USE_CLI_ENV_FILE
+            value: "true"
+          {{- end }}
+          # For running pod with read only file system we write all the cache files to /var/cache volume mount
+          - name: TUF_ROOT
```
Comment (barmagnezi) [inline wiz-admission-controller/templates/deploymentauditlogs.yaml:144]: We don't need cache dir for audit logs

File: wiz-admission-controller/Chart.yaml [hunk 2]
```diff
@@ -7,10 +7,10 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 3.4.13
+version: 3.4.13-alpha
```
Comment (barmagnezi) [inline wiz-admission-controller/Chart.yaml:10]:
```suggestion
version: 3.4.14-alpha
```

File: wiz-admission-controller/templates/deployment.yaml [hunk 3]
```diff
@@ -1,15 +1,17 @@
+{{- if and (or .Values.opaWebhook.enabled .Values.imageIntegrityWebhook.enabled .Values.debugWebhook.enabled)}}
```
Comment (barmagnezi) [inline wiz-admission-controller/templates/deployment.yaml:1]: why do you need and + or here?

File: wiz-admission-controller/templates/deploymentauditlogs.yaml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            name: server-certs
+            readOnly: true
+          {{- with .Values.customVolumeMounts }}
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- with .Values.global.customVolumeMounts }}
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+      volumes:
+        - name: cache
```
Comment (barmagnezi) [inline wiz-admission-controller/templates/deploymentauditlogs.yaml:176]: We don't need cache dir for audit logs

File: wiz-admission-controller/values.yaml [hunk 5]
```diff
@@ -151,6 +151,52 @@ imageIntegrityWebhook:
...
+        - "cronjobs"
+        - "replicasets"
+        - "statefulsets"
+        - "daemonsets"
+  namespaceSelector:  # https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-namespaceselector
+    matchExpressions:
+    - key: kubernetes.io/metadata.name # Using "kubernetes.io/metadata.name" key to exclude specific namespaces (https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2161-apiserver-default-labels/README.md#proposal)
+      operator: NotIn
+      values:
+      - kube-system
```
Comment (barmagnezi) [inline wiz-admission-controller/values.yaml:190]: Ar you sure we don't want to monitor reviews from kube system?

File: wiz-admission-controller/templates/opawebhook.yaml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 {{- $tlsCrt := .Values.tlsCertificate.tlsCertificate -}}
 {{- $tlsKey := .Values.tlsCertificate.tlsKey -}}
 {{- if .Values.tlsCertificate.create -}}
-{{- $altNames := list ( printf "%s.%s" (include "wiz-admission-controller.fullname" .) .Release.Namespace ) ( printf "%s.%s.svc" (include "wiz-admission-controller.fullname" .) .Release.Namespace ) -}}
+{{- $altNames := list ( printf "%s.%s" (include "wiz-admission-controller.fullname" .) .Release.Namespace ) ( printf "%s.%s.svc" (include "wiz-admission-controller.fullname" .) .Release.Namespace ) ( printf "%s.%s" ( include "wiz-admission-controller-audit-logs.name" .) .Release.Namespace ) ( printf "%s.%s.svc" ( include "wiz-admission-controller-audit-logs.name" .) .Release.Namespace ) -}}
```
Comment (barmagnezi) [inline wiz-admission-controller/templates/opawebhook.yaml:5]: Add the new service also to certmanager.yaml file

**Review by barmagnezi — CHANGES_REQUESTED** [review]

File: wiz-admission-controller/values.yaml (hunk 5 above)
Comment (nitzanzuler) [reply wiz-admission-controller/values.yaml:190]: Yes, https://wiz-sec.slack.com/archives/C05HCDVJMTN/p1710083199005799?thread_ts=1710078245.159749&cid=C05HCDVJMTN
And I verified this with him.

File: wiz-admission-controller/Chart.yaml [hunk 7]
```diff
@@ -7,10 +7,10 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 3.4.13
+version: 3.4.14-alpha
```
Comment (barmagnezi) [inline wiz-admission-controller/Chart.yaml:10]:
```suggestion
version: 3.5.0-alpha
```

**Review by barmagnezi — APPROVED** [review]

File: wiz-admission-controller/values.yaml [hunk 8]
```diff
@@ -151,6 +151,52 @@ imageIntegrityWebhook:
...
+        - "ingresses"
+        - "services"
+        - "networkpolicies"
+        - "pods"
+        - "deployments"
+        - "jobs"
+        - "cronjobs"
+        - "replicasets"
+        - "statefulsets"
+        - "daemonsets"
```
Comment (sshayb) [inline wiz-admission-controller/values.yaml:184]: There is this one deprecated controller called ReplicationController that can also serve as a workload, lets add it as well:
```
apiVersion: v1
kind: ReplicationController
metadata:
  name: nginx
spec:
  replicas: 3
  selector:
    app: nginx
  template:
    metadata:
      name: nginx
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx
        ports:
        - containerPort: 80
```

Comment (nitzanzuler) [reply wiz-admission-controller/values.yaml:184]: OK, added

**Pull request merged**